channeldb/migration12/migration_test.go

```diff
@@ -0,0 +1,306 @@
+		0x6c, 0x64, 0x09, 0x62, 0x79, 0x65, 0x20, 0x77, 0x6f, 0x72,
+		0x6c, 0x64, 0x06, 0x70, 0x61, 0x79, 0x72, 0x65, 0x71, 0x00,
+		0x00, 0x00, 0x20, 0x00, 0x00, 0x4e, 0x94, 0x91, 0x4f, 0x00,
+		0x00, 0x0f, 0x01, 0x00, 0x00, 0x00, 0x0e, 0x77, 0xc4, 0xd3,
+		0xd5, 0x00, 0x00, 0x00, 0x00, 0xfe, 0x20, 0x0f, 0x01, 0x00,
+		0x00, 0x00, 0x0e, 0x77, 0xd5, 0xc8, 0x1c, 0x00, 0x00, 0x00,
+		0x00, 0xfe, 0x20, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x03, 0xe8, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x05, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0xa4, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x41, 0x01, 0x08, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x08, 0x03, 0x08, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x09, 0x05, 0x08, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x64, 0x07, 0x04, 0x00, 0x00,
+		0x00, 0x58, 0x09, 0x08, 0x00, 0x13, 0xbc, 0xbf, 0x72, 0x4e,
+		0x1e, 0x00, 0x0b, 0x08, 0x00, 0x17, 0xaf, 0x4c, 0x22, 0xc4,
+		0x24, 0x00, 0x0d, 0x04, 0x00, 0x00, 0x23, 0x1d, 0x0f, 0x01,
+		0x02,
+	}
+
+	afterInvoice1Htlc = []byte{
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xb8, 0x01, 0x0b,
+		0x68, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c,
+		0x64, 0x03, 0x06, 0x70, 0x61, 0x79, 0x72, 0x65, 0x71, 0x05,
+		0x0f, 0x01, 0x00, 0x00, 0x00, 0x0e, 0x77, 0xc4, 0xd3, 0xd5,
+		0x00, 0x00, 0x00, 0x00, 0xfe, 0x20, 0x07, 0x0f, 0x01, 0x00,
+		0x00, 0x00, 0x0e, 0x77, 0xd5, 0xc8, 0x1c, 0x00, 0x00, 0x00,
+		0x00, 0xfe, 0x20, 0x09, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x05, 0x0b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x06, 0x0d, 0x20, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42,
+		0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x0f, 0x08, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0xe8, 0x11, 0x04, 0x00,
+		0x00, 0x00, 0x20, 0x13, 0x08, 0x00, 0x00, 0x4e, 0x94, 0x91,
+		0x4f, 0x00, 0x00, 0x15, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x17, 0x00, 0x19,
+		0x01, 0x03, 0x1b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+		0x01, 0xa4, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x41,
+		0x01, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x08,
+		0x03, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x09,
+		0x05, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x64,
+		0x07, 0x04, 0x00, 0x00, 0x00, 0x58, 0x09, 0x08, 0x00, 0x13,
+		0xbc, 0xbf, 0x72, 0x4e, 0x1e, 0x00, 0x0b, 0x08, 0x00, 0x17,
+		0xaf, 0x4c, 0x22, 0xc4, 0x24, 0x00, 0x0d, 0x04, 0x00, 0x00,
+		0x23, 0x1d, 0x0f, 0x01, 0x02,
+	}
```

Comment:
This is indeed an idea too, to prevent even more serialize code.

Reply:
yes this prevented me from having to copy in the old serialization code :)